src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTask.cs

```diff
@@ -80,22 +82,43 @@ public IPackTaskLogic PackTaskLogic
 
         public override bool Execute()
         {
-            var request = GetRequest();
-            var logic = PackTaskLogic;
-            PackageBuilder packageBuilder = null;
-            var packArgs = logic.GetPackArgs(request);
-
-            // If packing using a Nuspec file, we don't need to build a PackageBuilder here
-            // as the package builder is built by reading the manifest file later in the code path.
-            // Passing a null package builder for nuspec file code path is perfectly valid.
-            if (string.IsNullOrEmpty(request.NuspecFile))
+            try
             {
-                packageBuilder = logic.GetPackageBuilder(request);
+                var request = GetRequest();
+                var logic = PackTaskLogic;
+                PackageBuilder packageBuilder = null;
+                var packArgs = logic.GetPackArgs(request);
+
+                // If packing using a Nuspec file, we don't need to build a PackageBuilder here
+                // as the package builder is built by reading the manifest file later in the code path.
+                // Passing a null package builder for nuspec file code path is perfectly valid.
+                if (string.IsNullOrEmpty(request.NuspecFile))
+                {
+                    packageBuilder = logic.GetPackageBuilder(request);
+                }
+                var packRunner = logic.GetPackCommandRunner(request, packArgs, packageBuilder);
+                logic.BuildPackage(packRunner);
+
+                return true;
             }
-            var packRunner = logic.GetPackCommandRunner(request, packArgs, packageBuilder);
-            logic.BuildPackage(packRunner);
+            catch (Exception ex)
+            {
+                // Log the error
+                if (ExceptionLogger.Instance.ShowStack)
+                {
+                    Logger.LogError(ex.ToString());
+                }
+                else
+                {
+                    Logger.LogError(ExceptionUtilities.DisplayMessage(ex));
+                }
+
+                // Log the stack trace as verbose output.
+                Logger.LogVerbose(ex.ToString());
```

Comment:
log to LogDebug instead of LogVerbose, I think most of the code does it that way.